core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1177,8 +1177,9 @@ abstract class RDD[T: ClassTag](
    *                   It must be greater than 0.000017.
    */
   def countApproxDistinct(relativeSD: Double = 0.05): Long = withScope {
+    require(relativeSD > 0.000017, s"accuracy ($relativeSD) must be greater than 0.000017")
     val p = math.ceil(2.0 * math.log(1.054 / relativeSD) / math.log(2)).toInt
-    countApproxDistinct(p, 0)
+    countApproxDistinct(if (p < 4) 4 else p, 0)
```

Comment:
Seems OK, in that it enforces the contract in the docs, and then avoids an error in a case that does not seem to be an error. @rxin do you have any thoughts on this one?